fiftyone/utils/utils3d.py

```diff
@@ -417,6 +428,226 @@ def _compute_intersection_points(box1, box2):
+) -> np.ndarray:
+    """Computes the 3D corners of a cuboid given its location, rotation, and
+    dimensions.
+
+    Args:
+        location: a 3-element list or np.ndarray representing the (x, y, z)
+            location of the cuboid
+        rotation: a 3-element list or np.ndarray representing the (roll, pitch,
+            yaw) rotation in radians
+        dimension: a 3-element list or np.ndarray representing the (length,
+            width, height) of the cuboid
+
+    Returns:
+        A 3x8 np.ndarray containing the 3D coordinates of the cuboid's corners.
+    """
+    l, w, h = dimension
+    roll, pitch, yaw = rotation
+    rot = R.from_euler("zyx", [yaw, pitch, roll])  # yaw-pitch-roll order
+    rotation_matrix = rot.as_matrix()
+    x_corners = l / 2 * np.array([1, 1, 1, 1, -1, -1, -1, -1])
+    y_corners = w / 2 * np.array([1, -1, -1, 1, 1, -1, -1, 1])
+    z_corners = h / 2 * np.array([1, 1, -1, -1, 1, 1, -1, -1])
+    corners = np.vstack((x_corners, y_corners, z_corners))
+    corners = np.dot(rotation_matrix, corners)
+    location = np.array(location).reshape(3, 1)
+    corners += location
+    return corners
+
+
+def pinhole_projector(
+    points: np.ndarray, cam_params: Dict, normalize=True
+) -> np.ndarray:
+    """Projects 3D detection points to 2D using the given camera intrinsics
+    assuming a pinhole camera.
+
+    The following orientation is assumed- x axis points to the right in the
+    image plane, y axis points down in the image plane and z axis points forward
+    from the camera.
+
+    Args:
+        points: a 3xN np.ndarray containing the 3D coordinates of the points
+        cam_params: a dict containing the key 'intrinsics' that maps to a 3x3 or
+            4x4 np.ndarray representing the camera intrinsics matrix
+        normalize (True): whether to normalize the projected points by their
+            z-coordinate
+
+    Returns:
+        A 2xN np.ndarray containing the projected 2D coordinates of the points.
+        If `normalize` is True, the points are normalized by their z-coordinate.
+    """
+    camera_intrinsics = cam_params.get("intrinsics")
+    if camera_intrinsics is None:
+        raise ValueError("camera_params['intrinsics'] is required")
+    camera_intrinsics = np.asarray(camera_intrinsics)
+    if camera_intrinsics.shape[0] > 4 or camera_intrinsics.shape[1] > 4:
+        raise ValueError("intrinsics must be 3x3 or 4x4")
+    if points.shape[0] != 3:
+        raise ValueError("points must be a 3xN array")
+
+    cam_int_pad = np.eye(4)
+    cam_int_pad[
+        : camera_intrinsics.shape[0], : camera_intrinsics.shape[1]
+    ] = camera_intrinsics
+    nbr_points = points.shape[1]
+
+    points = np.concatenate((points, np.ones((1, nbr_points))))
+    points = np.dot(cam_int_pad, points)
+    points = points[:3, :]
+    if normalize:
+        points = points / points[2:3, :].repeat(3, 0).reshape(3, nbr_points)
+    return points[:2, :]
```

Comment:
should we consider culling points z<0 since they can't be projected to the image plane?